crates/bevy_gizmos/src/primitives/dim3.rs

```diff
@@ -0,0 +1,822 @@
+
+    fn primitive_3d(
+        &mut self,
+        primitive: Sphere,
+        position: Vec3,
+        rotation: Quat,
+        color: Color,
+    ) -> Self::Output<'_> {
+        SphereBuilder {
+            gizmos: self,
+            radius: primitive.radius,
+            position,
+            rotation,
+            color,
+            segments: DEFAULT_NUMBER_SEGMENTS,
+        }
+    }
+}
+
+impl<T: GizmoConfigGroup> Drop for SphereBuilder<'_, '_, '_, T> {
+    fn drop(&mut self) {
+        if !self.gizmos.enabled {
+            return;
+        }
+
+        let SphereBuilder {
+            radius,
+            position: center,
+            rotation,
+            color,
+            segments,
+            ..
+        } = self;
+
+        // draw two caps, one for the "upper half" and one for the "lower" half of the sphere
+        [-1.0, 1.0].into_iter().for_each(|sign| {
+            let top = *center + (*rotation * Vec3::Y) * sign * *radius;
+            draw_cap(
+                self.gizmos,
+                *radius,
+                *segments,
+                *rotation,
+                *center,
+                top,
+                *color,
+            );
+        });
+
+        draw_circle(self.gizmos, *radius, *segments, *rotation, *center, *color);
+    }
+}
+
+// plane 3d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive3d<Plane3d> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self: 'a;
+
+    fn primitive_3d(
+        &mut self,
+        primitive: Plane3d,
+        position: Vec3,
+        rotation: Quat,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        let normal = rotation * *primitive.normal;
+        self.arrow(position, position + normal, color);
+        let ortho = normal.any_orthonormal_vector();
+        (0..4)
+            .map(|i| i as f32 * 0.25 * TAU)
+            .map(|angle| Quat::from_axis_angle(normal, angle))
+            .for_each(|quat| {
+                let dir = quat * ortho;
+                (0..)
+                    .filter(|i| i % 2 == 0)
+                    .map(|i| [i, i + 1])
+                    .map(|percents| percents.map(|p| p as f32 * 0.25 * dir))
+                    .map(|vs| vs.map(|v| v + position))
+                    .take(3)
+                    .for_each(|[start, end]| {
+                        self.line(start, end, color);
+                    });
+            });
+    }
+}
+
+// line 3d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive3d<Line3d> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self: 'a;
+
+    fn primitive_3d(
+        &mut self,
+        primitive: Line3d,
+        position: Vec3,
+        rotation: Quat,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        let dir = rotation * *primitive.direction;
+        self.arrow(position, position + dir, color);
```

Comment:
Same as 2D
```suggestion
        let dir = rotation * *primitive.direction;
        self.arrow(position - dir, position, color);
```
Also, the naming and variables are slightly different from 2D (2D has `direction` and variables `start` and `end`)